skimage/segmentation/_join.py

```diff
@@ -120,14 +120,11 @@ def relabel_sequential(label_field, offset=1):
         label_field = label_field.astype(new_type)
         m = m.astype(new_type)  # Ensures m is an integer
     labels = np.unique(label_field)
-    labels0 = labels[labels != 0]
-    if m == len(labels0):  # nothing to do, already 1...n labels
-        return label_field, labels, labels
+    relabeled = np.digitize(label_field.reshape(-1, ), labels).reshape(label_field.shape)
```

Comment:
In addition to addressing the test failures, change `reshape(-1, )` to `ravel()`